NEPS/Hacks/Misc.cpp

```diff
@@ -1859,3 +1870,75 @@ void Misc::fakePrime() noexcept
 		#endif
 	}
 }
+
+void Misc::drawStartPos(ImDrawList* drawList, Vector& quickpeekstartpos) noexcept
+{
+	if (!localPlayer || !localPlayer->isAlive())
+		return;
+
+	if (quickpeekstartpos != Vector{ 0, 0, 0 }) {
+		ImVec2 startpos;
+		if (Helpers::worldToScreen(quickpeekstartpos, startpos))
+			drawList->AddCircleFilled(startpos, 25, ImColor(1.f, 1.f, 1.f, 1.f), 32);
+	}
+}
+
+void gotoStart(UserCmd * cmd) noexcept
+{
+	if (!localPlayer || !localPlayer->isAlive())
+		return;
+
+	if (usercmds_for_quickpeek.empty())
+		return;
+
+	if (hasShot)
+	{
+		if (qp_count > 0)
+		{
+			cmd->upmove = -usercmds_for_quickpeek.at(qp_count).upmove;
+			cmd->sidemove = -usercmds_for_quickpeek.at(qp_count).sidemove;
+			cmd->forwardmove = -usercmds_for_quickpeek.at(qp_count).forwardmove;
+			//cmd->viewangles = usercmds_for_quickpeek.at(qp_count).viewAngles;
+			//g_EngineClient->SetViewAngles(&cmd->viewangles);
+			qp_count--;
+		}
+	}
+	else
+	{
+		qp_count = usercmds_for_quickpeek.size();
+	}
+}

```

Comment:
Oh, I see you did the way I would, even though it's a bit more complex, you managed to do it right, my respect 👍🏿